stubs/networkx/networkx/algorithms/boundary.pyi

```diff
@@ -0,0 +1,65 @@
+from _typeshed import Incomplete
+from collections.abc import Iterable
+from typing import TypeVar, overload
+from typing_extensions import Literal
+
+from networkx.classes.graph import Graph
+
+_T = TypeVar("_T")
+_U = TypeVar("_U")
+
+@overload
+def edge_boundary(
+    G: Graph[_T],
+    nbunch1: Iterable[_T],
+    nbunch2: Iterable[_T] | None = ...,
+    data: Literal[False] = ...,
+    keys: Literal[False] = ...,
+    default: Incomplete = ...,
+) -> Iterable[tuple[_T, _T]]: ...
+@overload
+def edge_boundary(
+    G: Graph[_T],
+    nbunch1: Iterable[_T],
+    nbunch2: Iterable[_T] | None = ...,
+    data: Literal[True] = ...,
+    keys: Literal[False] = ...,
+    default: Incomplete = ...,
+) -> Iterable[tuple[_T, _T, dict[str, Incomplete]]]: ...
+@overload
+def edge_boundary(
+    G: Graph[_T],
+    nbunch1: Iterable[_T],
+    nbunch2: Iterable[_T] | None = ...,
+    data: str = ...,
+    keys: Literal[False] = ...,
+    default: _U | None = None,
+) -> Iterable[tuple[_T, _T, dict[str, _U]]]: ...
+@overload
+def edge_boundary(
+    G: Graph[_T],
+    nbunch1: Iterable[_T],
+    nbunch2: Iterable[_T] | None = ...,
+    data: Literal[False] = ...,
+    keys: Literal[True] = ...,
+    default: Incomplete = ...,
+) -> Iterable[tuple[_T, _T, int]]: ...
+@overload
+def edge_boundary(
+    G: Graph[_T],
+    nbunch1: Iterable[_T],
+    nbunch2: Iterable[_T] | None = ...,
+    data: Literal[True] = ...,
+    keys: Literal[True] = ...,
+    default: Incomplete = ...,
+) -> Iterable[tuple[_T, _T, int, dict[str, Incomplete]]]: ...
+@overload
+def edge_boundary(
+    G: Graph[_T],
+    nbunch1: Iterable[_T],
+    nbunch2: Iterable[_T] | None = ...,
+    data: str = ...,
+    keys: Literal[True] = ...,
+    default: _U | None = None,
+) -> Iterable[tuple[_T, _T, int, dict[str, _U]]]: ...
```

Comment:
https://github.com/python/typeshed/actions/runs/6042836622/job/16398777431?pr=10544#step:5:183
```
stubs/networkx/networkx/algorithms/boundary.pyi:12: error: Overloaded function signatures 1 and 2 overlap with incompatible return types  [misc]
stubs/networkx/networkx/algorithms/boundary.pyi:12: error: Overloaded function signatures 1 and 3 overlap with incompatible return types  [misc]
stubs/networkx/networkx/algorithms/boundary.pyi:12: error: Overloaded function signatures 1 and 4 overlap with incompatible return types  [misc]
stubs/networkx/networkx/algorithms/boundary.pyi:12: error: Overloaded function signatures 1 and 5 overlap with incompatible return types  [misc]
stubs/networkx/networkx/algorithms/boundary.pyi:12: error: Overloaded function signatures 1 and 6 overlap with incompatible return types  [misc]
stubs/networkx/networkx/algorithms/boundary.pyi:21: error: Overloaded function signatures 2 and 4 overlap with incompatible return types  [misc]
stubs/networkx/networkx/algorithms/boundary.pyi:21: error: Overloaded function signatures 2 and 5 overlap with incompatible return types  [misc]
stubs/networkx/networkx/algorithms/boundary.pyi:21: error: Overloaded function signatures 2 and 6 overlap with incompatible return types  [misc]
stubs/networkx/networkx/algorithms/boundary.pyi:30: error: Overloaded function signatures 3 and 4 overlap with incompatible return types  [misc]
stubs/networkx/networkx/algorithms/boundary.pyi:30: error: Overloaded function signatures 3 and 5 overlap with incompatible return types  [misc]
stubs/networkx/networkx/algorithms/boundary.pyi:30: error: Overloaded function signatures 3 and 6 overlap with incompatible return types  [misc]
stubs/networkx/networkx/algorithms/boundary.pyi:39: error: Overloaded function signatures 4 and 5 overlap with incompatible return types  [misc]
stubs/networkx/networkx/algorithms/boundary.pyi:39: error: Overloaded function signatures 4 and 6 overlap with incompatible return types  [misc]
```

```suggestion
@overload
def edge_boundary(
    G: Graph[_T],
    nbunch1: Iterable[_T],
    nbunch2: Iterable[_T] | None = None,
    data: Literal[False] = False,
    keys: Literal[False] = False,
    default=None,
) -> Iterable[tuple[_T, _T]]: ...
@overload
def edge_boundary(
    G: Graph[_T],
    nbunch1: Iterable[_T],
    nbunch2: Iterable[_T] | None,
    data: Literal[True],
    keys: Literal[False] = False,
    default=None,
) -> Iterable[tuple[_T, _T, dict[str, Incomplete]]]: ...
@overload
def edge_boundary(
    G: Graph[_T],
    nbunch1: Iterable[_T],
    nbunch2: Iterable[_T] | None = None,
    *,
    data: Literal[True],
    keys: Literal[False] = False,
    default=None,
) -> Iterable[tuple[_T, _T, dict[str, Incomplete]]]: ...
@overload
def edge_boundary(
    G: Graph[_T],
    nbunch1: Iterable[_T],
    nbunch2: Iterable[_T] | None,
    data: str,
    keys: Literal[False] = False,
    default: _U | None = None,
) -> Iterable[tuple[_T, _T, dict[str, _U]]]: ...
@overload
def edge_boundary(
    G: Graph[_T],
    nbunch1: Iterable[_T],
    nbunch2: Iterable[_T] | None = None,
    *,
    data: str,
    keys: Literal[False] = False,
    default: _U | None = None,
) -> Iterable[tuple[_T, _T, dict[str, _U]]]: ...
@overload
def edge_boundary(
    G: Graph[_T], nbunch1: Iterable[_T], nbunch2: Iterable[_T] | None, data: Literal[False], keys: Literal[True], default=None
) -> Iterable[tuple[_T, _T, int]]: ...
@overload
def edge_boundary(
    G: Graph[_T],
    nbunch1: Iterable[_T],
    nbunch2: Iterable[_T] | None = None,
    data: Literal[False] = False,
    *,
    keys: Literal[True],
    default=None,
) -> Iterable[tuple[_T, _T, int]]: ...
@overload
def edge_boundary(
    G: Graph[_T], nbunch1: Iterable[_T], nbunch2: Iterable[_T] | None, data: Literal[True], keys: Literal[True], default=None
) -> Iterable[tuple[_T, _T, int, dict[str, Incomplete]]]: ...
@overload
def edge_boundary(
    G: Graph[_T],
    nbunch1: Iterable[_T],
    nbunch2: Iterable[_T] | None = None,
    *,
    data: Literal[True],
    keys: Literal[True],
    default=None,
) -> Iterable[tuple[_T, _T, int, dict[str, Incomplete]]]: ...
@overload
def edge_boundary(
    G: Graph[_T], nbunch1: Iterable[_T], nbunch2: Iterable[_T] | None, data: str, keys: Literal[True], default: _U | None = None
) -> Iterable[tuple[_T, _T, int, dict[str, _U]]]: ...
@overload
def edge_boundary(
    G: Graph[_T],
    nbunch1: Iterable[_T],
    nbunch2: Iterable[_T] | None = None,
    *,
    data: str,
    keys: Literal[True],
    default: _U | None = None,
) -> Iterable[tuple[_T, _T, int, dict[str, _U]]]: ...
```

This one is unclear, I believe it's partly caused by the TypeVar, but filling in default values fixes it.